src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -235,7 +235,7 @@ class ObjectMonitor : public CHeapObj<mtObjectMonitor> {
   // to the ObjectMonitor reference manipulation code:
   //
   #define OM_OFFSET_NO_MONITOR_VALUE_TAG(f) \
-    ((in_bytes(ObjectMonitor::f ## _offset())) - markWord::monitor_value)
+    ((in_bytes(ObjectMonitor::f ## _offset())) - (int)markWord::monitor_value)
```

Comment:
```suggestion
    ((in_bytes(ObjectMonitor::f ## _offset())) - (intptr_t)markWord::monitor_value)
```

Reply:
If this is intptr_t, it'll get a Wconversion warning on the uses that expect int.

Reply:
OK, in that case checked_cast<int>() would be more correct.

Reply:
Ok.  Added a checked_cast.